File: src/main/java/skywolf46/commandannotation/annotations/autocomplete/AutoComplete.java
package skywolf46.commandannotation.annotations.autocomplete;

//import skywolf46.commandannotation.autocompleter.OnlinePlayerCompleter;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(value = RetentionPolicy.RUNTIME)
@Target(value = ElementType.METHOD)
public @interface AutoComplete {
    String value() default "";

//    Class<? extends AbstractAutoCompleterProvider> provider() default OnlinePlayerCompleter.class;
}

File: core/src/main/kotlin/skywolf46/commandannotation/kotlin/annotation/AutoComplete.kt
package skywolf46.commandannotation.kotlin.annotation

/**
 * Provide autocomplete to command.
 * Because autocomplete will be based on system structure, some autocomplete features will not work. (Like discord)
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class AutoComplete()